.github/workflows/release.yaml

```diff
@@ -0,0 +1,578 @@
+name: Release process
+on:
+  # Releases can be manually triggered if the version has already been updated and the apidocs were
+  # properly built
+  workflow_dispatch:
+
+  # Releases are also allowed to be automatically triggered once the release preparation 
+  # pull-request (targeting main) is merged. Here, we also make sure that the PR has changes 
+  # in at least one file related to `make set_version` or `make apidocs`
+  pull_request:
```

Comment:
it will be automatically triggered when a closed PR targetting main that changed at least something in the `version.py` file or in the apidocs file.

Reply:
Not so sure about this one tbh.
I'm fine with it but this should be discussed with the team imo (fine for now though)

Reply:
Alright this is to trigger the packaging process after a release preparation right?

Reply:
yes !